runtime/compiler/runtime/MethodMetaData.c

```diff
@@ -2127,15 +2134,21 @@ UDATA getCurrentByteCodeIndexAndIsSameReceiver(J9TR_MethodMetaData * methodMetaD
          prevInlinedCallSite = inlinedCallSite;
          inlinedCallSite = getNextInlinedCallSite(methodMetaData, inlinedCallSite);
          }
-      byteCodeInfo = (TR_ByteCodeInfo *)getByteCodeInfo(prevInlinedCallSite);
-      if (inlinedCallSite)
+      if (prevInlinedCallSite)
+         byteCodeInfo = (TR_ByteCodeInfo *)getByteCodeInfo(prevInlinedCallSite);
+      else if (inlinedCallSite)
          byteCodeInfo = (TR_ByteCodeInfo *)getByteCodeInfo(inlinedCallSite);
 
       }
 
-   if (isSameReceiver != 0)
-      *isSameReceiver = byteCodeInfo->_isSameReceiver;
-   return byteCodeInfo->_byteCodeIndex;
+   if (byteCodeInfo)
+      {
+      if (isSameReceiver != 0)
+         *isSameReceiver = byteCodeInfo->_isSameReceiver;
+      return byteCodeInfo->_byteCodeIndex;
+      }
+
+   return UDATA_MAX;
```

Comment:
How do you know this is an acceptable return value for this function?

Reply:
If we end up with an invalid `byteCodeInfo`, then the function does not have a valid bytecode index to return. `UDATA_MAX` in this case represents returning some invalid value, as bytecode index 0 could be treated as valid by the caller. I chose this based on how bytecode index is set to `UDATA_MAX` when it cannot be determined such as in `iterateStackTrace`, which is also where this function is used:

https://github.com/eclipse-openj9/openj9/blob/1c6ce868dd4ddf285fa13100dfe5ac391bf5834d/runtime/vm/exceptiondescribe.c#L306-L311